crates/iota-storage/src/http_key_value_store.rs

```diff
@@ -85,6 +86,169 @@ pub enum Key {
     EventsByTxDigest(TransactionDigest),
 }
 
+impl Key {
+    pub const TRANSACTION: &'static str = "tx";
+    pub const TX_EFFECTS: &'static str = "fx";
+    pub const CHECKPOINT_CONTENTS: &'static str = "cc";
+    pub const CHECKPOINT_SUMMARY: &'static str = "cs";
+    pub const TX2C: &'static str = "tx2c";
+    pub const OBJECT: &'static str = "ob";
+    pub const EVTX: &'static str = "evtx";
+
+    // Create a [`Key`] instance based on the provided item type and
+    /// [`base64_url`] encoded string.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use std::str::FromStr;
+    ///
+    /// use iota_storage::http_key_value_store::Key;
+    /// use iota_types::digests::TransactionDigest;
+    ///
+    /// let key = Key::new("tx", "7jb54RvJduLj9HdV9L41UJqZ5KWdzYY2rl1eL8AVl9o").unwrap();
+    /// assert_eq!(
+    ///     key,
+    ///     Key::Tx(
+    ///         TransactionDigest::from_str("H2tetNL3CfroDF3iJNA7wFo6oRQiJedGTeykZi6HAGqP").unwrap()
+    ///     )
+    /// );
+    /// ```
+    pub fn new(item_type: &str, encoded_key: &str) -> anyhow::Result<Self> {
+        let decoded_key = base64_url::decode(encoded_key)
+            .map_err(|err| anyhow::anyhow!("invalid base64 url string: {err}"))?;
+
+        match item_type {
```

Comment:
Seems like a good place for an enum, unless I'm missing something?

Reply:
I made a PR in case you prefer this: https://github.com/iotaledger/iota/pull/5406